cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
Yeah you're right here. Damn this one is tricky. Couldn't we be doing the following:

Public and private are reserved keys. Only one key can have the key-id. On, for example, key rotation, the keys with kid public and private get reassigned something like `private:<timestamp/uuid>` and the new keys get public/private?